### PR TITLE
WN: certs /5

### PR DESCRIPTION
**Pull request by Rick-Anderson**
## Internal review content: 
[Trust the ASP.NET Core HTTPS development certificate on Linux](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?view=aspnetcore-8.0&branch=pr-en-us-33564)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/release-notes/aspnetcore-9.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/7408d1038e8ef04261ae8a202ab9c356ae97eb01/aspnetcore/release-notes/aspnetcore-9.0.md) | [What's new in ASP.NET Core 9.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?branch=pr-en-us-33564) |

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@amcasey  CoPilot made a bunch of additions and they looked helpful to me.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#### Trust the ASP.NET Core HTTPS development certificate on Linux
+
+On Ubuntu and Fedora based Linux distros, `dotnet dev-certs https --trust` now configures ASP.NET Core HTTPS development certificate as a trusted certificate for:
+
+* Chromium browsers, for example, Google Chrome, Microsoft Edge, Chromium, and Opera.
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:5]: I don't believe Opera is Chromium-based.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* Mozilla Firefox and other Mozilla browsers.
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:6]: I think they only make Firefox - probably "derived browsers"?

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+To establish trust in OpenSSL, the `dev-certs` tool:
+
+* Puts the certificate in `~/.aspnet/dev-certs/trust`
+* Runs a simplified version of OpenSSL's c_rehash tool on the directory.
+* Asks the user to update the `SSL_CERT_DIR` environment variable.
+
+To establish trust in dotnet, the tool puts the certificate in the `My/Root` certificate store.
+
+To establish trust in [NSS databases](https://docs.redhat.com/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), the tool:
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:19]: This link doesn't appear to work.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+* Searches the home directory for Firefox profiles.
+* For each profiles found, adds an entry to the `nssdb` database.
+
+The home directories searched are `~/.pki/nssdb` and `~/snap/chromium/current/.pki/nssdb`. The `~/.pki/nssdb` directory is used by Firefox and other Mozilla browsers. The `~/snap/chromium/current/.pki/nssdb` directory is used by Chromium browsers installed via snap.
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:24]: No, `~/.pki/nssdb` is used by Chromium.

**Review by amcasey — CHANGES_REQUESTED** [review]
I think I'm probably missing something, but this seems like a less accurate version of the text [here](https://github.com/dotnet/release-notes-drafts/pull/98/files#diff-c6a3a4bfa48460f3e111837755c6f6fbf252cb09505daf6822e246e44fbf1c2bR105).

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md (hunk 3 above)
Comment (Rick-Anderson) [reply aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md]: Arg, I removed the  local `/en` and didn't test. When  the  `/en`  local is used we like to remove them and normally it redirects to  `/en` .

**Comment by Rick-Anderson** [comment]
> I think I'm probably missing something, but this seems like a less accurate version of the text [here](https://github.com/dotnet/release-notes-drafts/pull/98/files#diff-c6a3a4bfa48460f3e111837755c6f6fbf252cb09505daf6822e246e44fbf1c2bR105).

I can review and do an almost straight copy of text [here](https://github.com/dotnet/release-notes-drafts/pull/98/files#diff-c6a3a4bfa48460f3e111837755c6f6fbf252cb09505daf6822e246e44fbf1c2bR105) if you prefer. Some of the sentences are a bit long.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md (hunk 1 above)
Comment (amcasey) [reply aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md]: Apparently, they rebased it on Chromium, but I'd still rather not list it.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* Asks the user to update the `SSL_CERT_DIR` environment variable.
+
+To establish trust in dotnet, the tool puts the certificate in the `My/Root` certificate store.
+
+To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), the tool:
+
+* Searches the home directory for Firefox profiles.
+* For each profiles found, adds an entry to the `nssdb` database.
+
+The home directories searched are `~/.pki/nssdb` and `~/snap/chromium/current/.pki/nssdb`.
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:27]: This reads as though it describes where Firefox profiles are sought but it actually describes which folders we look for to support Chromium.

**Comment by amcasey** [comment]
> > I think I'm probably missing something, but this seems like a less accurate version of the text [here](https://github.com/dotnet/release-notes-drafts/pull/98/files#diff-c6a3a4bfa48460f3e111837755c6f6fbf252cb09505daf6822e246e44fbf1c2bR105).
> 
> I can revert and do an almost straight copy of text [here](https://github.com/dotnet/release-notes-drafts/pull/98/files#diff-c6a3a4bfa48460f3e111837755c6f6fbf252cb09505daf6822e246e44fbf1c2bR105) if you prefer. Some of the sentences are a bit long.

I don't feel strongly either way - I just thought it was worth pointing out that we had similar text elsewhere.

**Comment by Rick-Anderson** [comment]
@amcasey  I misinterpreted your version. I have two versions now, a slight change to your's (I changed  for each one found to for each directory found)

Select which version you like better. When possible, shorter sentences are easier for English speakers to parse and vastly easier for MT (machine translation)

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), the tool searches the home directory for Firefox profiles, *~/.pki/nssdb*, and *~/snap/chromium/current/.pki/nssdb*. For each directory found, the tool adds an entry to the nssdb.
+
+<!--
+To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), the tool searches the relevant directories.  For each directory found, the tool adds an entry to the nssdb.
+
+The relevant directories are:
+
+* The home directory for Firefox profiles.
+* `~/.pki/nssdb`, and `~/snap/chromium/current/.pki/nssdb` for Chromium browsers.
+-->
```
Comment (Rick-Anderson) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:31]: Original vs. my rewrite, pick the winner or suggest something different

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md (hunk 1 above)
Comment (Rick-Anderson) [reply aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md]: > Apparently, they rebased it on Chromium, but I'd still rather not list it.

I deleted Chromium

Comment (amcasey) [reply aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md]: Sorry, my comment was unclear - I was following up about Opera, which apparently _is_ Chromium-based now.  The text was fine after removing Opera from the list:

> * Chromium browsers, for example, Google Chrome, Microsoft Edge, and Chromium.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md (hunk 6 above)
Comment (amcasey) [reply aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:30]: Neither is particularly elegant.  What about something like:

To establish trust in browsers, the tool adds entries to the [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), if any, in the current user's Firefox profiles, *~/.pki/nssdb*, and *~/snap/chromium/current/.pki/nssdb*.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+To establish trust in OpenSSL, the `dev-certs` tool:
+
+* Puts the certificate in `~/.aspnet/dev-certs/trust`
+* Runs a simplified version of OpenSSL's [c_rehash tool](https://docs.openssl.org/1.0.2/man1/c_rehash/) on the directory.
+* Asks the user to update the `SSL_CERT_DIR` environment variable.
+
+To establish trust in dotnet, the tool puts the certificate in the `My/Root` certificate store.
+
+To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), if any, in the current user's Firefox profiles, ~/.pki/nssdb, and ~/snap/chromium/current/.pki/nssdb.
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:22]: This doesn't seem to say what is done to establish trust.

File: aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md [hunk 8]
```diff
@@ -19,4 +19,4 @@ To establish trust in OpenSSL, the `dev-certs` tool:
 
 To establish trust in dotnet, the tool puts the certificate in the `My/Root` certificate store.
 
-To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), if any, in the current user's Firefox profiles, ~/.pki/nssdb, and ~/snap/chromium/current/.pki/nssdb.
\ No newline at end of file
+To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), if any, the tool searches the home directory for Firefox profiles, `~/.pki/nssdb`, and `~/snap/chromium/current/.pki/nssdb`. For each directory found, the tool adds an entry to the `nssdb`
```
Comment (amcasey) [inline aspnetcore/release-notes/aspnetcore-9/includes/trust_dev_cert_linux.md:22]:
```suggestion
To establish trust in [NSS databases](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/developer_guide/che-nsslib), if any, the tool searches the home directory for Firefox profiles, `~/.pki/nssdb`, and `~/snap/chromium/current/.pki/nssdb`. For each directory found, the tool adds an entry to the `nssdb`.
```

**Review by amcasey — APPROVED** [review]

**Pull request merged**